src/Microsoft.IdentityModel.Tokens/Validation/IssuerValidationResult.cs

```diff
@@ -11,17 +11,27 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     internal class IssuerValidationResult : ValidationResult
     {
+        internal enum ValidationSource
```

Comment:
@iNinja  I think this is ok to move forward while everything is internal, but we will need a more general solution as users will need to set their own ValidationSource.

ValidationFailureType has an extensibility model that may shed light on a solution.

Reply:
Thanks. I will make a note to adopt a model based on `ValidationFailureType` as we progress.